### DISCUSSION TITLE
Ecto error when installing app to store

### DISCUSSION THREAD

**Discussion by stiegi** [ericdude4/shopifex#64] [Q&A] [answered]
Hi, first of all thank you very much for your work on this!

I tried this with fresh Phoenix 1.6.16 and 1.7.3: after following the install instructions and setting everything up (with mix shopifex.install), during Shop install I get this error message:

```
Ecto.QueryError at GET /initialize-installation
deps/shopifex/lib/shopifex/shops.ex:24: field `url` in `where` does not exist in schema TestApp.ShopifyShops.ShopifyShop in query:

from s0 in TestApp.ShopifyShops.ShopifyShop,
  where: s0.url == ^"https://foobar.myshopify.com/",
  select: s0
```

And sure enough, that table is empty. I was wondering if there is another step I have to do before I try to install the plugin to a store? To populate the table I mean.

**Comment by ericdude4** [comment, marked as answer]
hi @stiegi !

Sorry about that! I think the `mix shopifex.install` no longer injects the fields into the schemas properly any more, so you need to manually update the following files and input their field names: `shopify_shop.ex`, `shopify_grant.ex`, `shopify_plan.ex`.

For example, these changes were needed for `shopify_shop.ex`:

## Before:
```elixir
defmodule ShopifexTest.ShopifyShops.ShopifyShop do
  use Ecto.Schema
  import Ecto.Changeset
  
  schema "shopify_shops" do
    
    has_many :grants, ShopifexTest.ShopifyShops.ShopifyGrant, foreign_key: :shop_id
    
    timestamps()
  end

  @doc false
  def changeset(shopify_shop, attrs) do
    shopify_shop
    |> cast(attrs, [:url, :access_token, :scope])
    |> validate_required( [:url, :access_token, :scope])
  end
end
```

## After:
```elixir
defmodule ShopifexTest.ShopifyShops.ShopifyShop do
  use Ecto.Schema
  import Ecto.Changeset
  
  schema "shopify_shops" do
    field :url, :string
    field :access_token, :string
    field :scope, :string
    
    has_many :grants, ShopifexTest.ShopifyShops.ShopifyGrant, foreign_key: :shop_id
    
    timestamps()
  end

  @doc false
  def changeset(shopify_shop, attrs) do
    shopify_shop
    |> cast(attrs, [:url, :access_token, :scope])
    |> validate_required( [:url, :access_token, :scope])
  end
end
```

You can determine which fields are missing from each schema based on the changeset's `cast/3` call.

I haven't had time to fix this, but it would be a great opportunity for someone to contribute to the library :)

**Reply by stiegi** [reply]
Thank you very much for your quick response, this made it work!
